portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -33,42 +35,135 @@
 /** Servlet that encapsulates comments. */
 @WebServlet("/data")
 public final class DataServlet extends HttpServlet {
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment");
+    PreparedQuery results = fetch(); 
+    List<Comment> comments = toInternal(results);
+    String userFilter = getLastFilter(comments);
```

Comment:
The solution with the passing the sorting order "as a comment"─creative, but also a bit "hacky"─does not address a potential requirement that clients should be able to ask for a particular sorting order, independently from each other.

Have you considered sorting on the client? Then the server doesn't need to care about sorting at all and the server doesn't need to care which user wants what kind of sorting. Server-side sorting matters however once the total number of comments is so large that you can no longer send all of the comments to the client and have to sort first, and then send the first N comments to the client.